javaagent-tooling/build.gradle.kts

```diff
@@ -41,6 +41,8 @@ dependencies {
 
   implementation("io.opentelemetry.contrib:opentelemetry-aws-xray-propagator")
 
+  implementation("org.apache.commons:commons-text:1.10.0")
```

Comment:
We try to keep the dependencies as minimal as possible; parsing a `${placeholder}` should be reasonably easy to do inline, without depending on a 3rd party library.

Reply:
I _can_ write code to to this inline, but I will not be able to do as well as [the Apache commons-text implementation](https://github.com/apache/commons-text/blob/master/src/main/java/org/apache/commons/text/StringSubstitutor.java#L1404), and there may be bugs 😅 

(Apache commons-text has [escapes](https://github.com/apache/commons-text/blob/master/src/main/java/org/apache/commons/text/StrSubstitutor.java#L934), [defaults values](https://github.com/apache/commons-text/blob/master/src/main/java/org/apache/commons/text/StrSubstitutor.java#L147),  recursive substitutions etc)

WDYT?

Reply:
> (Apache commons-text has [escapes](https://github.com/apache/commons-text/blob/master/src/main/java/org/apache/commons/text/StrSubstitutor.java#L934), [defaults values](https://github.com/apache/commons-text/blob/master/src/main/java/org/apache/commons/text/StrSubstitutor.java#L147), recursive substitutions etc)

That is actually a good reason _not_ to include commons-text -- it has way too many features, and it historically had security vulnerabilities inside the text interpolation engine.

A naive `indexOf()`-based implementation should be enough for this project for the time being.
(Also FYI there is a broader initiative of a general yaml-based OpenTelemetry file configuration format that's being worked on right now, while it'll be at least a couple months before it's actually used anywhere it's going to replace the java-specific properties at some point.)

Reply:
Okay I will have a look, but it has just occurred to me that we will at least have to handle escaping:

It is entirely possible that someone has a literal value which looks like `${...}`, which they don't want to be substituted; for example someone may be configuring a header with a security token (e.g., `otel.exporter.otlp.headers`) which may look like a substitution placeholder.

I will also add some java docs and tests so the substitutor contract is clear.

🫠

Reply:
Oh, and so I guess this PR could be a breaking change for someone unlucky enough to end up with an unexpected substitution...

Reply:
> Okay I will have a look, but it has just occurred to me that we will at least have to handle escaping:

Hm, you're right, that's a valid scenario -- we'll have to implement some simple escaping too.

> Oh, and so I guess this PR could be a breaking change for someone unlucky enough to end up with an unexpected substitution...

I think the chance of that happening is very low, wouldn't worry too much about that.